usr/sbin/registercloudguest

```diff
@@ -287,6 +289,72 @@ def get_responding_update_server(region_smt_servers):
     sys.exit(1)
 
 
+# ----------------------------------------------------------------------------
+def get_product_tree(product_file='/etc/products.d/baseproduct'):
+    """
+    Read product element from baseproduct and return an etree
+    """
+    if os.path.isfile(product_file):
+        with open(product_file, 'r') as product:
+            raw_xml = product.read()
+            product_entry_point = raw_xml.index('<product')
+            return etree.fromstring(raw_xml[product_entry_point:])
+
+
+# ----------------------------------------------------------------------------
+def get_product_triplet(product_tree):
+    """
+    Extract identifier, version and arch information from
+    the given product tree
+    """
+    return [
+        product_tree.find('name').text.lower(),
+        product_tree.find('version').text,
+        product_tree.find('arch').text
+    ]
+
+
+# ----------------------------------------------------------------------------
+def setup_ltss_registration(registration_target, regcode):
+    """
+    Run SUSEConnect to register LTSS for this instance
+    """
+    logging.info('Running LTSS registration...')
+    product = get_product_tree()
+    if product is None:
+        message = 'Cannot find baseproduct registration for LTSS'
+        logging.error(message)
+        print(message, file=sys.stderr)
+        sys.exit(1)
+
+    ltss_registered = False
+    if os.path.isdir(ZYPP_SERVICES):
+        for entry in os.listdir(ZYPP_SERVICES):
+            if 'LTSS' in entry:
+                ltss_registered = True
+                break
+    if ltss_registered:
+        logging.info('LTSS already registered')
+    else:
+        name, version, arch = get_product_triplet(product)
```

Comment:
But the name will be `SLES` for example and when we run `SUSEConnect the name should be `SLES-LTSS` because we are trying to register the LTSS extension, right?

Reply:
Oh right, I completely forgot to update the name properly. Thanks

Reply:
done, and I added a comment about our current strategy